test/functional/p2p_permissions.py

```diff
@@ -146,7 +146,14 @@ def check_tx_relay(self):
             [tx],
```

Comment:
nit: Could add a comment here that this first relay is only needed to make the node think it already has the transaction

(sorry for leaving nits in succession instead of in one reply, feel free to ignore)

Reply:
keep them coming!  I'm going to ignore this one, I'll reconsider it if there are more changes.